### DISCUSSION TITLE
PnP Modern Search - 4.8.0

### DISCUSSION THREAD

**Discussion by FranckyC** [microsoft-search/pnp-modern-search#2588] [Announcements]
**This solution includes only one `.sppkg` file:**
- `pnp-modern-search-parts-<version_number>.sppkg` (Mandatory).

> Extensions (i.e. custom web components, layouts, etc.) must be deployed separately. See [PnP Modern Search - Extensibility samples](https://github.com/microsoft-search/pnp-modern-search-extensibility-samples) for more information.

**Useful resources**

- [Official documentation](https://microsoft-search.github.io/pnp-modern-search/)

____

## What's Changed
* Fix for main branch package.json by @wobba in https://github.com/microsoft-search/pnp-modern-search/pull/2201
* Fixed #2205 - The custom layouts form extensibility libraries were not loaded in the property pane. by @FranckyC in https://github.com/microsoft-search/pnp-modern-search/pull/2218
* Fixed #1967 <mgt-person> and DetailsList z-index by @FranckyC in https://github.com/microsoft-search/pnp-modern-search/pull/2219
* Fixed #2211 - Revert to previous version of @pnp/spfx-property-controls (3.2.0) by @FranckyC in https://github.com/microsoft-search/pnp-modern-search/pull/2220
* Fixed #1941 - Add "Presence.Read.All" permissions by default for the SPFx package by @FranckyC in https://github.com/microsoft-search/pnp-modern-search/pull/2252
* Scroll to view top of results by @emilyk0520 in https://github.com/microsoft-search/pnp-modern-search/pull/2244
* Bump terser from 4.8.0 to 4.8.1 in /search-extensibility by @dependabot in https://github.com/microsoft-search/pnp-modern-search/pull/2246
* Fixed #2212 regarding localization flag and owstaxidmetadataalltagsinfo managed property by @FranckyC in https://github.com/microsoft-search/pnp-modern-search/pull/2267
* Fixed #2280 - URL hash change was not triggered for search box by @FranckyC in https://github.com/microsoft-search/pnp-modern-search/pull/2285
* Fixed #2259 - Reset page number when new filters are selected by @FranckyC in https://github.com/microsoft-search/pnp-modern-search/pull/2286
* Bump terser from 4.8.0 to 4.8.1 in /search-extensibility by @dependabot in https://github.com/microsoft-search/pnp-modern-search/pull/2300
* Added missing translations for the sort control (#2288) by @FranckyC in https://github.com/microsoft-search/pnp-modern-search/pull/2301
* Disable PnP js telemetry by @wobba in https://github.com/microsoft-search/pnp-modern-search/pull/2309
* Fix #2308 by @wobba in https://github.com/microsoft-search/pnp-modern-search/pull/2312
* news scenario by @davemehr in https://github.com/microsoft-search/pnp-modern-search/pull/2323
* Added support for actionable Adaptive Cards in results by @PaoloPia in https://github.com/microsoft-search/pnp-modern-search/pull/2243
* Fixed small issues and typos in docs by @PaoloPia in https://github.com/microsoft-search/pnp-modern-search/pull/2361
* Added Teams messages entity type option for Microsoft search data source by @FranckyC in https://github.com/microsoft-search/pnp-modern-search/pull/2364
* Creating pt-br localization files and correct en-us file. by @Porantim in https://github.com/microsoft-search/pnp-modern-search/pull/2370
* Introducing clear search on empty search box by @salascz in https://github.com/microsoft-search/pnp-modern-search/pull/2369
* Fix to not use input query text after disabling by @patrikhellgren in https://github.com/microsoft-search/pnp-modern-search/pull/2395
* Disable interleaving when using query rules. by @wobba in https://github.com/microsoft-search/pnp-modern-search/pull/2473
* Fix: Rerender on empty results when inputQuery changes by @edarroudi in https://github.com/microsoft-search/pnp-modern-search/pull/2462
* Beta Entity Types Bookmark & Acronym by @edarroudi in https://github.com/microsoft-search/pnp-modern-search/pull/2446
* Upgrade spfx 115 by @wobba in https://github.com/microsoft-search/pnp-modern-search/pull/2518
* Pagination deeplink by @ferrarirosso in https://github.com/microsoft-search/pnp-modern-search/pull/2503
* Add "semver": "7.3.5" back by @brianpmccullough in https://github.com/microsoft-search/pnp-modern-search/pull/2528
* Use icons from fluentui package by @wobba in https://github.com/microsoft-search/pnp-modern-search/pull/2531
* Feature: interface query transformation by @edarroudi in https://github.com/microsoft-search/pnp-modern-search/pull/2035
* Enable custom data source extensibility by @patrikhellgren in https://github.com/microsoft-search/pnp-modern-search/pull/2400
* Removed a Typo by @asadrefai in https://github.com/microsoft-search/pnp-modern-search/pull/2254
* [Fix] Filter count bug due to PR2503 by @edarroudi in https://github.com/microsoft-search/pnp-modern-search/pull/2548
* Fix #2483 by @wobba in https://github.com/microsoft-search/pnp-modern-search/pull/2554
* Fixed #2555 - Added token resolution in the refinement filters field for SharePoint data source by @FranckyC in https://github.com/microsoft-search/pnp-modern-search/pull/2560
* Added Microsoft Graph Toolkit option for adaptive cards layout by @FranckyC in https://github.com/microsoft-search/pnp-modern-search/pull/2569

## New Contributors
* @emilyk0520 made their first contribution in https://github.com/microsoft-search/pnp-modern-search/pull/2244
* @PaoloPia made their first contribution in https://github.com/microsoft-search/pnp-modern-search/pull/2243
* @Porantim made their first contribution in https://github.com/microsoft-search/pnp-modern-search/pull/2370
* @salascz made their first contribution in https://github.com/microsoft-search/pnp-modern-search/pull/2369
* @ferrarirosso made their first contribution in https://github.com/microsoft-search/pnp-modern-search/pull/2503
* @brianpmccullough made their first contribution in https://github.com/microsoft-search/pnp-modern-search/pull/2528
* @asadrefai made their first contribution in https://github.com/microsoft-search/pnp-modern-search/pull/2254

**Full Changelog**: https://github.com/microsoft-search/pnp-modern-search/compare/4.7.0...4.8.0

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft-search/pnp-modern-search/releases/tag/4.8.0'>PnP Modern Search - 4.8.0</a>.</em>

**Comment by brianpmccullough** [comment]
@wobba @FranckyC - I confirmed the following today in a tenant:

4.6.1 tenant deployed.  Activated app on site1.  As expected, I verified I could add web parts using the page authoring tools (i.e. web parts were available from gallery with page in edit mode).  

4.7.0 subsequently tenant deployed.  Activated app on site2.  I verified that the search web parts were not available upon activation in Site2 (they remained activated and available in Site1).

4.8.0 subsequently tenant deployed (this version).    Activated app on Site3.  I verified I could add web parts using the page authoriing tools on Site3, I could still NOT on Site2.  I then removed the app from Site2, and then added it back.  I confirmed that I was then able to add the Search web parts in Site2 using the page authoring tools.

So, to summarize, it looks like 4.8.0 does appear to resolve the feature activation issue from the added feature node which was subsequently removed with this update: https://github.com/microsoft-search/pnp-modern-search/commit/9d8584ac951d8bf26e088eba7d75990fd45591f0

**Reply by wobba** [reply]
Thank you for verifying :)

**Comment by jltitus225** [comment]
I'm currently still using version 1.0. Are there directions somewhere on how to upgrade my version to 4.8?

**Reply by kasperbolarsen** [reply]
As per the documentation there is not upgrade path from version 3.* to 4.*
You will have to scan you tenant for search web parts and replace those which are needed.
I am pretty sure that I once made a PowerShell script that added the App on the sites and added the web parts on specific pages.  Such a script will make the upgrade a wee bit less painful.
